### DISCUSSION TITLE
Test epoch 1.16 mining

### DISCUSSION THREAD

**Discussion by leeyemi** [tearust/teaproject#16] [Q&A] [answered]
Hi,

1. My assets and leader board have different information in term of tea account balance and staking debt. The reason is because leader board is converted to USD. My suggestion is to show the information defaulted to Tea and Coffee and there will be a button to convert the view from Tea & Coffee to USD
2. If possible please add the principal and interest under my collateral seeds and the contract term, as this is a main page for user to monitor their loan and interest.
3. What is the genesis loan shown on the leader board? Is it correct if I assumed this is the total principal and interest of the loan using the cml seeds as collateral? 
4. May I know the reason we cannot listed our cml under collateral for auctions? My thought process is the smart contract can regulate condition. If cml under collateral is sold then there will be automatic execution pay off debt (principal and interest). The miner who sold the cml will get the TEA after deduction of above.
5. In related to auction, it is right now a free world, everyone can determine whatever the price for any type of cml they listed in. My concern is this page will be spammed with lots of overpriced listing. And the main page of market place does not show the type of cml (A/B/C), the performance of the CM and also the slot which currently staked unless we click the detail one by on. For buyer such information is very useful to determine participating on the bidding. And we don't want to waste time clicking each listing. Suggest if we could have such information and sorting / filter ability.

 
![image](https://user-images.githubusercontent.com/88189869/129522592-42a3ec19-707b-4dea-a353-57707281ae39.png)
![image](https://user-images.githubusercontent.com/88189869/129522656-f60d7e8b-7a6e-4ff1-adae-0af90fc46780.png)

**Comment by kevingzhang** [comment, marked as answer]
> My assets and leader board have different information in term of tea account balance and staking debt. The reason is because leader board is converted to USD. My suggestion is to show the information defaulted to Tea and Coffee and there will be a button to convert the view from Tea & Coffee to USD

You helped us find a bug. It should be COFFEE, not USD. I log a new issue

https://github.com/tearust/tea-browser-wallet/issues/191

**Reply by kevingzhang** [reply]
We can change to a radio button say "Showing in COFFEE now. Change to TEA?"
Is this OK?

**Comment by geopor** [comment]
- For questions 2 & 3 related to CML collateralized loans, there's an issue of more information being shown on leader board and not enough information being shown on the CML collateral page. We should have a better designed page for that in coming updates.
- Regarding 4 about auctioning CML collateral, the loans are an early stage benefit that will be phased out just as the marketplace starts picking up steam. Introducing such logic for CML seed loans that will be phased out is not a development priority.
- Regarding 5, this would be a good idea. I will raise an issue to see if it can be included in future updates.

**Comment by kevingzhang** [comment]
> In related to auction, it is right now a free world, everyone can determine whatever the price for any type of cml they listed in. My concern is this page will be spammed with lots of overpriced listing. 

The spam issue has been taking care in our design, just did not show in the UI.
Whenever you create a new auction, you will asked to pay a deposit and a fee. The fee will be taken every new trading window starts. The spammer will lost the fee every 1000 blocks (current trading window interval). 

> And the main page of market place does not show the type of cml (A/B/C), the performance of the CM and also the slot which currently staked unless we click the detail one by on. For buyer such information is very useful to determine participating on the bidding. And we don't want to waste time clicking each listing. 


Adding CML type doesn't solve the problem. You would have to click the CML to see the detail. The type of CML can tell the basic information, but every CML is different (They are NFT), so there are more information you will need to know before taking a bid, such as:
- Age
- Lifespan
- Peak performance
- Current staking situation
- Current performance
- Debt?
- Lien?

We cannot put all this information on the table. So clicking the CML to get detail is probably unavoidable. 


> Suggest if we could have such information and sorting / filter ability.

This is a good point and doable.

**Comment by kevingzhang** [comment]
> If possible please add the principal and interest under my collateral seeds and the contract term, as this is a main page for user to monitor their loan and interest.


the principal and interest rate are not constant, they are changeable. The DAO can vote to change the number. So it is hard to have variables in the contract text.

What do you think if we show the number at the header of the collateral page? Let me know if you agree and we can create an issue on this. @leeyemi

**Reply by leeyemi** [reply]
What do you think if we show the number at the header of the collateral page? Let me know if you agree and we can create an issue on this. @leeyemi

I think so, it would be easier to monitor the loan rather that going to leader board.

**Comment by kevingzhang** [comment]
> What is the genesis loan shown on the leader board? Is it correct if I assumed this is the total principal and interest of the loan using the cml seeds as collateral?

Yes, you are right

**Comment by kevingzhang** [comment]
> May I know the reason we cannot listed our cml under collateral for auctions? My thought process is the smart contract can regulate condition. If cml under collateral is sold then there will be automatic execution pay off debt (principal and interest). The miner who sold the cml will get the TEA after deduction of above.

This is a future function we will develop in a standalone product called "Golden Bridge Loan" tApp. 
The Genesis Loan is free tApp and it is part of the blockchain. We do not want it to be too complicate and fully featured so that other commercial tApps won't survive in the competition. TEA Project is a platform to get more developers to build their own application.  If have you watched our strategic 3 steps video, you will know what I am talking about.

In our design, the Genesis Loan will take care the basic features, leave the opportunities to other developers and their commercial applications.
